--- redir-2.2.1/Makefile.orig	Fri Feb 28 20:37:30 2003
+++ redir-2.2.1/Makefile	Fri Feb 28 20:38:38 2003
@@ -9,8 +9,8 @@
 # if you would like support for TCP wrappers (and have libwrap.a
 # installed), remove these comments.
 
-WRAP_CFLAGS = # -DUSE_TCP_WRAPPERS 
-WRAP_LIBS = # -lwrap
+WRAP_CFLAGS =  -DUSE_TCP_WRAPPERS 
+WRAP_LIBS =  -lwrap
 
 # if your system needs any additional libraries (solaris, for example, 
 # needs the ones commented out below), edit this line.
@@ -32,8 +32,8 @@
 # if your system lacks getopt_long, remove the comment from this line
 OBJS = redir.o $(GETOPT_OBJS)
 
-CFLAGS = -O2 -Wall $(STR_CFLAGS) $(WRAP_CFLAGS) $(EXTRA_CFLAGS)
-LDFLAGS = -s
+CFLAGS = $(OPT_FLAGS) -Wall -DUSE_TCP_WRAPPERS # -DNEED_STRRCHR -DNEED_STRDUP
+LDFLAGS = -g
 
 # solaris, and others, may also need these libraries to link
 # also edit here if you're using the TCP wrappers code
--- redir-2.2.1/redir.c.orig	Fri Feb 28 20:38:55 2003
+++ redir-2.2.1/redir.c	Fri Feb 28 20:41:34 2003
@@ -72,6 +72,7 @@
 #include <syslog.h>
 #include <sys/types.h>
 #include <sys/socket.h>
+#include <time.h>
 #include <sys/time.h>
 #include <sys/wait.h>
 #include <netinet/in.h>
@@ -510,7 +511,7 @@
 		perror("getsockname");
 		if (dosyslog)
 			syslog(LOG_ERR, "getsockname failed: %m");
-		exit(1);
+		return;
 	}
 
 	lport = ntohs(sockname.sin_port);
@@ -745,7 +746,7 @@
 	}
      
 	debug1("peer IP is %s\n", inet_ntoa(client.sin_addr));
-	debug1("peer socket is %d\n", client.sin_port);
+	debug1("peer socket is %d\n", ntohs(client.sin_port));
 
 	/*
 	 * Double fork here so we don't have to wait later
@@ -871,8 +872,8 @@
 		strcpy(tmp2, inet_ntoa(target->sin_addr));
 	  
 		syslog(LOG_NOTICE, "connecting %s/%d to %s/%d",
-		       tmp1, client.sin_port,
-		       tmp2, target->sin_port);
+                       tmp1, ntohs(client.sin_port),
+                       tmp2, ntohs(target->sin_port));
 	}
 
 	/* do proxy stuff */
@@ -1066,7 +1067,7 @@
 
 		if (!getpeername(0, (struct sockaddr *) &client, &client_size)) {
 			debug1("peer IP is %s\n", inet_ntoa(client.sin_addr));
-			debug1("peer socket is %d\n", client.sin_port);
+			debug1("peer socket is %d\n", ntohs(client.sin_port));
 		}
 		if ((targetsock = socket(AF_INET, SOCK_STREAM, 0)) < 0) {
 			perror("target: socket");
@@ -1109,8 +1110,8 @@
 
 		if (dosyslog) {
 			syslog(LOG_NOTICE, "connecting %s/%d to %s/%d",
-			       inet_ntoa(client.sin_addr), client.sin_port,
-			       target_ip, target.sin_port);
+			       inet_ntoa(client.sin_addr), ntohs(client.sin_port),
+			       target_ip, ntohs(target.sin_port));
 		}
 
 		/* Just start copying - one side of the loop is stdin - 0 */
